client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -74,13 +143,35 @@ const CancelPurchaseButton = React.createClass( {
 		} );
 	},
 
+	handleSubmit( error, response ) {
+		if ( error ) {
+			notices.error( error.message );
+			return;
+		}
+
+		notices.success( response.message, { persistent: true } );
+
+		analytics.tracks.recordEvent(
+			'calypso_purchases_cancel_form_submit',
+			{ product_slug: this.props.purchase.productSlug }
+		);
+
+		page.redirect( paths.list() );
+	},
+
+	submitCancelPurchase() {
+		wpcom.cancelAndRefundPurchase( this.props.purchase.id, null, ( error, response ) => {
```

Comment:
Super minor, but this could just be

```
wpcom.cancelAndRefundPurchase( this.props.purchase.id, null, this.handleSubmit );
```

`error` should be undefined if `response` is present, so we probably don't need `error || null`.